yt_dlp/extractor/rokfin.py

```diff
@@ -254,3 +422,91 @@ def _real_extract(self, url):
         return self.playlist_result(
             self._entries(channel_id, channel_name, self._TABS[tab]),
             f'{channel_id}-{tab}', f'{channel_name} - {tab.title()}', str_or_none(channel_info.get('description')))
+
+
+# E.g.: rkfnsearch5:"\"zelenko\"" or rkfnsearch5:"\"mollie james\""
+class RokfinSearchIE(SearchInfoExtractor):
+    IE_DESC = 'Rokfin Search'
+    _TYPES = {
+        'video': (('id', 'raw'), 'post'),
+        'audio': (('id', 'raw'), 'post'),
+        'stream': (("content_id", 'raw'), 'stream'),
+        'dead_stream': (('content_id', 'raw'), 'stream'),
+        'stack': (('content_id', 'raw'), 'stack'),
+    }
+    IE_NAME = 'rokfin:search'
+    _SEARCH_KEY = 'rkfnsearch'
+    _NETRC_MACHINE = False
+    _BASE_URL = 'https://rokfin.com'
+    _CACHE_SECTION_NAME = 'rokfin-search-engine-access'
+    _search_engine_access_url = None
+    _search_engine_access_key = None
+
+    def _real_initialize(self):
+        self._search_engine_access_url = self._downloader.cache.load(self._CACHE_SECTION_NAME, 'url')
+        self._search_engine_access_key = self._downloader.cache.load(self._CACHE_SECTION_NAME, 'key')
+        if not self._search_engine_access_url or not self._search_engine_access_key:
+            self._search_engine_access_url, self._search_engine_access_key = self._get_search_engine_access_credentials()
+
+    def _search_results(self, query):
+        def get_video_data(metadata):
+            for search_result in metadata.get('results', []):
+                video_id_ind, video_type = self._TYPES.get(traverse_obj(search_result, ('content_type', 'raw')), (None, None))
```

Comment:
what's `ind`?

Reply:
`video_id_ind` should have been called `video_id_key`.